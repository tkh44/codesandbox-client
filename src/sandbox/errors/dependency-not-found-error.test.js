import DependencyError from './dependency-not-found-error';

test('it handles normal deps correctly', () => {
  const error = new DependencyError('redux-form');

  expect(error.message).toMatchSnapshot();
});

test('it parses dependency names with slashes correctly', () => {
  const error = new DependencyError('redux-form/immutable');

  expect(error.message).toMatchSnapshot();
});

test('it parses dependency names with multiple slashes correctly', () => {
  const error = new DependencyError('redux-form/immutable/koekjes');

  expect(error.message).toMatchSnapshot();
});

test('it parses scoped packages', () => {
  const error = new DependencyError('@vx/group');

  expect(error.message).toMatchSnapshot();
});
