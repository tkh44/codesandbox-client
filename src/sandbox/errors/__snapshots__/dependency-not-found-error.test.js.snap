// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it handles normal deps correctly 1`] = `"Could not find dependency: 'redux-form'"`;

exports[`it parses dependency names with multiple slashes correctly 1`] = `"Could not find dependency: 'redux-form'"`;

exports[`it parses dependency names with slashes correctly 1`] = `"Could not find dependency: 'redux-form'"`;

exports[`it parses scoped packages 1`] = `"Could not find dependency: '@vx/group'"`;
